lib/thredded/email_notifier.rb

```diff
@@ -0,0 +1,53 @@
+module Thredded
+  class EmailNotifier
+
+    def new_post(post, users)
+      @post = post
+      users_opted_in = exclude_those_opting_out_of_followed_activity_notifications(users)
+      return if users_opted_in.empty?
+      PostMailer.post_notification(post.id, users_opted_in.map(&:email)).deliver_now
+      MembersMarkedNotified.new(post, users_opted_in).run
+    end
+
+    def new_private_post(post, users)
+      @post = post
+      @topic = post.postable
+      users = private_topic_recipients
+      return unless users.present?
+      user_emails = users.map(&:email)
+      PrivateTopicMailer
+        .message_notification(post.postable.id, user_emails)
+        .deliver_later
+      MembersMarkedNotified.new(post, users).run
+    end
+
+    def private_topic_recipients
+      users = @topic.users - [@post.user]
+      users = exclude_those_opting_out_of_message_notifications(users)
+      users = exclude_previously_notified(users)
+      users
+    end
+
+    private
+    attr_reader :post, :topic
+
+    def exclude_those_opting_out_of_followed_activity_notifications(users)
+      users.select do |user|
+        user.thredded_user_preference.followed_topic_emails &&
+          user.thredded_user_messageboard_preferences.in(post.messageboard).followed_topic_emails
+      end
+    end
+
+    def exclude_those_opting_out_of_message_notifications(users)
```

Comment:
I'd rename this "exclude_those_opting_out_of_private_topic_notifications" then it makes it clearer